src/components/RegisterModal/index.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { FC } from 'react';
+import { Modal } from 'react-overlays';
+import { Text } from '@shared/Text';
+import { Button } from '@shared/Button';
+import { ButtonCancel } from '@shared/ButtonCancel';
+import { nanoid } from 'nanoid';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import {
+  RegisterInputsType,
+  RegisterModalPropsType,
+} from '@/components/RegisterModal/type';
+import { getCurrentDate } from '@/helpers/getCurrentDate';
+import { useForceUpdate } from '@/hooks/useForceUpdate';
+import { avatarNone, coverNone } from '@/constants/images';
+import { getDataFromLS } from '@/helpers/getStartUsers';
+
+export const RegisterModal: FC<RegisterModalPropsType> = ({
+  renderBackdrop,
+  loginOpen,
+  close,
+  isOpen,
+}) => {
+  const id = nanoid(10);
+  const forceUpdate = useForceUpdate();
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<RegisterInputsType>({
+    defaultValues: {
+      username: '',
+      password: '',
+      location: '',
+      firstName: '',
+      lastName: '',
+    },
+  });
+
+  const onSubmit: SubmitHandler<RegisterInputsType> = ({
```

Comment:
Same thing, too much logic in the component, I stated that I want you guys to have all this logic in useTwitter hook, components shouldn't know from where and how he get/store/update data, it would be very hard for you to change localstorage to back-end with the current implementation. Let's move all that kind of logic to useTwitter hook